File: src/Portfolio.Client/src/app/models/portfolio-metrics.ts
import { EnrichedAssetHolding } from './enriched-asset-holding';

export interface PortfolioMetrics {
    holdings: EnrichedAssetHolding[];
    totalPortfolioValueUsd: number;
    totalCostBasisUsd: number;
    netDepositsUsd: number;
    totalUnrealizedProfitLossUsd: number;
    totalRealizedProfitLossUsd: number;
    totalProfitLossUsd: number;
    unrealizedProfitLossPercentage: number;
    totalReturn: number;
}

File: src/Portfolio.Domain/ValueObjects/PortfolioMetrics.cs
namespace Portfolio.Domain.ValueObjects;

public class PortfolioMetrics
{
    public List<EnrichedAssetHolding> Holdings { get; set; } = new();
    public decimal TotalPortfolioValueUsd { get; set; }
    public decimal TotalCostBasisUsd { get; set; }
    public decimal NetDepositsUsd { get; set; }
    public decimal TotalUnrealizedProfitLossUsd { get; set; }
    public decimal TotalRealizedProfitLossUsd { get; set; }
    public decimal TotalProfitLossUsd { get; set; }
    public decimal UnrealizedProfitLossPercentage { get; set; }
    public decimal TotalReturn { get; set; }
}
